### PR TITLE
Implemented PWA

### PR DESCRIPTION
**Pull request by alistair3149**
Implemented requirements needed for Progressive Web App.
ServiceWorker is working but needed more work on caching API JSON files so that the whole site is usable in offline mode. But for now it is good as a start.

### CODE REVIEW

**Comment by alistair3149** [comment]
Yes the service worker cached the pages and their assets that you visited before. So let's say that I only visited the Carrack page and I went offline, I'll be able to visit the Carrack page again with the data there, and other ship pages as well as they use the same assets, but no data will be there as it's never retrieved.

As for the second question, the API age is separated so it can be set lowered to 5 mins and the rest for longer.

**Comment by alistair3149** [comment]
I updated the SW to check to fetch from network first and use the cache version after a timeout of 15 seconds (it can be lower to increase responsiveness, but depends on the fetch time of the API usually). I also switched the max age to 3 days instead, the old max age was a copy and paste from the SO and I didn't think that though >:(

**Comment by richardthombs** [comment]
Can you separate out the non-service-worker related commits into a different pull request? If I merge this, I have no choice but to also suck in all the UI changes and I'm not ready to do that yet :)

**Comment by alistair3149** [comment]
There is no UI change in this PR, all the UI changes are in #15 .

File: website/ngsw-config.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  ],
+  "dataGroups": [
+    {
+      "name": "api-freshness",
+      "urls": [
+        "https://scunpacked.com/api/**"
+      ],
+      "cacheConfig": {
+        "strategy": "freshness",
+        "maxSize": 100,
```
Comment (richardthombs) [inline website/ngsw-config.json]: I think 100 is too small. If you load the Carrack up while the dev tools are open, there are 74 requests. Something small like the Argo has 26. So I think maybe 20 ships * 50 requests per ship, we should have 1,000 as a minimum.

Comment (alistair3149) [reply website/ngsw-config.json]: Increased to 1,000 in the latest commit

**Comment by richardthombs** [comment]
I don't know if you are seeing the same data, but this is what I see in the PR:

image

**Comment by alistair3149** [comment]
All the commit are already merged into the master branch with the other PR, the duplicated one is because I rebased the branch. For the second commit, I changed the theme color to deeppink from the angular blue as that seems to be the main color on the site, it does not change the UI other than the browser and status bar color when the site is loaded on mobile.

**Comment by alistair3149** [comment]
Closed for now, will reopen after cleaning up the commit

**Comment by alistair3149** [comment]
Cleaned up commit and re-opened PR at #16

**Pull request closed without merging**